wrangler-core/src/main/java/io/cdap/wrangler/registry/UserDirectiveRegistry.java

```diff
@@ -112,28 +116,25 @@ public UserDirectiveRegistry(StageContext context) {
   @Nullable
   @Override
   public DirectiveInfo get(String namespace, String name) throws DirectiveLoadException {
-    if (!registry.containsKey(namespace)) {
-      return null;
-    }
-    Map<String, DirectiveInfo> namespaceRegistry = registry.get(namespace);
+    Class<? extends Directive> directive = null;
     try {
-      if (!namespaceRegistry.containsKey(name)) {
-        if (context != null) {
-          Class<? extends Directive> directive = context.loadPluginClass(name);
-          if (directive == null) {
-            throw new DirectiveLoadException(
-              String.format("10-5 - Unable to load the user defined directive '%s'. " +
-                              "Please check if the artifact containing UDD is still present. It was there when the " +
-                              "pipeline was deployed, but don't seem to find it now.", name)
-            );
-          }
-          DirectiveInfo classz = new DirectiveInfo(DirectiveInfo.Scope.USER, directive);
-          namespaceRegistry.put(classz.name(), classz);
-          return classz;
-        }
+      if (context != null) {
+        directive = context.loadPluginClass(name);
       } else {
-        return namespaceRegistry.get(name);
+        if (manager != null) {
+          PluginConfigurer configurer = manager.createPluginConfigurer(namespace);
```

Comment:
I believe this is a fairly expensive operation. Do you notice slowness when many directives are used? Though I guess it's probably similar to the previous implementation that loads all plugin artifacts.

Reply:
I did see a bit of slowness as we don't cache it anymore. I was thinking of caching it at this point to make sure it's faster in subsequent operations, but it happens only once so I decided not to do it.

Reply:
The correct thing would be for the service endpoint to create one PluginConfigurer, and for that same instance to be used throughout the lifetime of the call. This is a bigger refactor so I've opened https://issues.cask.co/browse/CDAP-15454 to revisit it later.